python/lib/dependabot/python/version.rb

```diff
@@ -16,7 +16,7 @@ class Version < Gem::Version
 
       # See https://peps.python.org/pep-0440/#appendix-b-parsing-version-strings-with-regular-expressions
       VERSION_PATTERN = 'v?([1-9][0-9]*!)?[0-9]+[0-9a-zA-Z]*(?>\.[0-9a-zA-Z]+)*' \
-                        '(-[0-9A-Za-z-]+(\.[0-9a-zA-Z-]+)*)?' \
+                        '(-[0-9A-Za-z]+(\.[0-9a-zA-Z]+)*)?' \
```

Comment:
```suggestion
                        '([0-9A-Za-z]+(\.[0-9a-zA-Z]+)*)?' \
```
Does this address the issue you mentioned internally?  [PEP 440](https://peps.python.org/pep-0440/#examples-of-compliant-version-schemes) has no valid examples with dashes so I think this gets even closer to the spec.

Reply:
From trying to figure https://peps.python.org/pep-0440/#appendix-b-parsing-version-strings-with-regular-expressions out, dash is only allowed to precede an explicitly list of pre|post|dev release identifiers. For example, "1.8-alpha" would be a valid version string while "1.8-arfa" would not.

Reply:
@pavera : We need to have a dash in the pre/post/dev as per https://peps.python.org/pep-0440/#pre-release-separators

```
Pre-releases should allow a ., -, or _ separator between the release segment and the pre-release segment.
The normal form for this is without a separator. This allows versions such as 1.1.a1 or 1.1-a1 which would be 
normalized to 1.1a1. It should also allow a separator to be used between the pre-release signifier and the numeral. 
This allows versions such as 1.0a.1 which would be normalized to 1.0a1.
```